interview-questions.md

```diff
@@ -7,63 +7,80 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a group of statements that preform a certain task.
 
-  Researched answer:
+  Researched answer: Functions are reusable pieces of code that are usually called upon a lot. They also
+  can have information pushed through what we call an argument.
 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: A .map goes through an array and returns one with the same length. A .filter
+  filters out some type of value.
 
-  Researched answer:
+  Researched answer: So .map goes through an array and returns another one with the
+  the same length, .filter does almost the same thing but only returns the
+  ones with true statement that you decided.
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log(), is like your "go" button because without it a lot of arrays won't be able to run.
+  A return is used when you input something into you console but will only return what you tell it to.
 
-  Researched answer:
+  Researched answer:console.log(), is almost you could say a go button because it
+  itll print things to the console so it can start the code. A return wont start
+  your console and only return something you put into it.
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument in a function is a way to pass information in the function.
 
-  Researched answer:
+  Researched answer: An argument contains outside information that a function needs in order to execute properly.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Its when one person is writing the code while the other can try to help with useful information
```

Comment:
So lets take this like a person is asking you this question in an interview. They asked you about the techniques but they also asked you what are the roles? You answered a bit of what pair programming is, but you didn't mention much about drivers, or navigators.